src/posit/connect/oauth.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from requests import Response, Session
+from typing import Optional
+
+from . import urls
+from .config import Config
+
+
+class OAuthIntegration:
+
+    def __init__(
+        self, config: Config, session: Session
+    ) -> None:
+        self.url = urls.append_path(config.url, "v1/oauth/integrations/credentials")
+        self.config = config
+        self.session = session
+
+
+    def get_credentials(self, user_identity: Optional[str]) -> Response:
```

Comment:
Does this need to return `Response` or can it return the response body or something? I don't see this being used anywhere yet, so I can't tell.

I think we're trying to avoid having the request/response details leaking outside of the get/put/post/etc. methods.

Reply:
Ah sorry, I was using github's symbol lookup and it didn't find it, should have stuck with good ol' cmd-F. 

Yeah in that case, I'd have this method return `response.json()`

Reply:
A defined type would be even better so that the consumer gets type hints.

```python
@dataclass
class OAuthResponse:
    access_token: str
```

Reply:
With the expectation that this endpoint might eventually return something other than an `access_token`, I like the idea of returning a defined type.